specification/security_model.adoc

```diff
@@ -1,9 +1,323 @@
 [[security_model]]
-== Security and Threat Model
+== Security Model
+
+The CoVE-IO securiy model is built on the following assumptions:
+
+- Host software components like the VMM or the hypervisor are not part of a TVM
+  TCB and are not trusted by TVMs.
+- Only the TVM owner can assess of a TDI trustworthiness. Based on that
+  assessment, it explicitly accepts or rejects a TDI into its TCB.
+- A TDI may access a TVM confidential memory through DMA only when all the
+  following conditions are met:
+  * The TVM owner has explicitly allowed the TDI to access its confidential
+    memory by accepting it.
+  * The TDI is exclusively assigned to the TVM, i.e. it must not be shared
+    other TVMs or any host software component.
+- A TVM may access a TDI trusted MMIO space only when all the following
+  conditions are met:
+  * The TVM owner has explicitly accepted the TDI.
+  * The TDI is exclusively assigned to the TVM, i.e. it must not be shared
+    other TVMs or any host software component.
+- Until a TDI is accepted by the TVM:
+  * The TDI is not allowed to DMA into the TVM confidential memory.
+  * Trusted MMIO access to the TDI is blocked by TSM.
+  * Untrusted MMIO access to the TDI may still be allowed by the VMM for
+    ordinary VMs (if and only if the TDI is in the TDISP unlocked state).
+
+By means of the CoVE-IO guest ABI, TVMs are required to explicitly accept
+TDIs into their TCBs. Accepting or rejecting a TDI is a TVM specific decision,
+based on TVM specific set of verification policies and criteria. A TVM accepting
+a TDI does not imply that other TVMs on the same host platform would accept it
+as well.
+
+Multi-function, CoVE-IO-compliant devices may expose multiple TDIs, assigned to
+different TVMs. It is the DSM responsibility to guarantee isolation between all
+assigned TDIs, on a per-TVM basis.
+
+Each TDI must be exclusively assigned to no more than one TVM. However, a single
+TVM can simultaneously be assigned several TDIs, irrespective of whether they
+originate from the same physical device or not.
+
+As a TVM TCB does not include PCIe switches and bridges, a selective PCIe IDE
+stream must be setup to guarantee end-to-end confidentiality and integrity
+protection between a TVM and a TDI. The host VMM is responsible for setting up a
+single selective IDE stream between the host platform and any physical device
+from which one or more TDIs are assigned to TVMs. All TDIs within that physical
+device share the single PCIe IDE stream.
```

Comment:
Should clarify that the host VMM is not in the TCB for the selective IDE stream - the endpoints for the selective IDE stream keys are configured by the TCB elements (TSM, RoT).

Reply:
Added, thanks.